### PR TITLE
Match height tabs bug

### LINKED ISSUE #9678: Match-height on tabs generates console error
#### How to reproduce this bug:

1. Create a tabstrip using the instructions at https://foundation.zurb.com/sites/docs/tabs.html
2. Assign `data-match-height=true` to the tabstrip's parent element
3. Load page with the tabstrip in a browser.

#### What should happen:

Tab panels should all scale to match the height of the tallest panel.

#### What happened instead:
- Tabs fail to match height.
- Standard tab behavior is disabled.
- Error message in console:
```javascript
TypeError: Cannot read property 'panelActiveClass' of undefined
    at HTMLDivElement.<anonymous> (foundation.min.js:4)
    at Function.each (jquery.min.js:2)
    at r.fn.init.each (jquery.min.js:2)
    at e.value (foundation.min.js:4)
    at e.value (foundation.min.js:4)
    at new e (foundation.min.js:4)
    at HTMLUListElement.<anonymous> (foundation.min.js:1)
    at Function.each (jquery.min.js:2)
    at r.fn.init.each (jquery.min.js:2)
    at String.<anonymous> (foundation.min.js:1)
```

#### Test case:

- [CodePen with Foundation 6.3, full CSS and Javascript libraries](http://codepen.io/wonkeythemonkey/pen/RKKOKE)

#### Source of bug:

This is a bug in the `_setHeight()` private function in `foundation-sites/js/foundation.tabs.js`

Here is the current code for the function, with comments to indicate where the problem lies.

```javascript
_setHeight() {
    var max = 0;
    this.$tabContent
      .find(`.${this.options.panelClass}`) // Incorrect instance of `this` -- current `this` has no `options`
      .css('height', '')
      .each(function() {
        var panel = $(this),
            isActive = panel.hasClass(`${this.options.panelActiveClass}`); // Incorrect instance of `this` -- current `this` has no `options`

        if (!isActive) {
          panel.css({'visibility': 'hidden', 'display': 'block'});
        }

        var temp = this.getBoundingClientRect().height;

        if (!isActive) {
          panel.css({
            'visibility': '',
            'display': ''
          });
        }

        max = temp > max ? temp : max;
      })
      .css('height', `${max}px`);
  }
```

#### Suggested fix:
```javascript
_setHeight() {
    var max = 0,
        _this = this; // Lock down the `this` value for the root tabs object

    this.$tabContent
      .find(`.${_this.options.panelClass}`) // get the options from the parent instead of trying to get them from the child
      .css('height', '')
      .each(function() {

        var panel = $(this),
            isActive = panel.hasClass(`${_this.options.panelActiveClass}`); // get the options from the parent instead of trying to get them from the child
						
        if (!isActive) {
          panel.css({'visibility': 'hidden', 'display': 'block'});
        }

        var temp = this.getBoundingClientRect().height;

        if (!isActive) {
          panel.css({
            'visibility': '',
            'display': ''
          });
        }

        max = temp > max ? temp : max;
      })
      .css('height', `${max}px`);
  }
```

### PR DESCRIPTION
**Pull request by IamManchanda**
Fixes #9678 & #9677 
@ncoden @kball for review 

Special Thanks: @wonkeythemonkey for **reporting** and the **code**.
Have tested it. works perfect now

------------------------------------------------------------------------------------------------------------------

Commits of #9659 is coming here too but as this is a correct branch for bugfix ( develop ) ...... so i guess i did the right thing.

### CODE REVIEW

File: js/foundation.tabs.js [hunk 1]
```diff
@@ -318,18 +318,22 @@ class Tabs {
...
    * @private
    */
   _setHeight() {
-    var max = 0;
+    var max = 0,
+        _this = this; // Lock down the `this` value for the root tabs object
+
     this.$tabContent
-      .find(`.${this.options.panelClass}`)
+      .find(`.${_this.options.panelClass}`) // get the options from the parent instead of trying to get them from the child
```
Comment (kball) [inline js/foundation.tabs.js]: I don't think this is necessary - this line is being evaluated in parent scope

**Comment by kball** [comment]
Noted one change that was unnecessary, other than that this looks good.

**Comment by IamManchanda** [comment]
done @kball . 
Also plz close #9677 on merge as i forgot to tag that for closure in my commits

**Comment by kball** [comment]
Merged! Thanks @IamManchanda

**Comment by IamManchanda** [comment]
Always pleasure kevin sir

**Pull request merged**